modules/plot/src/plot.cpp

```diff
@@ -0,0 +1,324 @@
+}
+
+
+void Plot::constructorHelper(cv::Mat Xdata, cv::Mat Ydata, const char * FigureName, int FigureWidth, int FigureHeight, double XAxisMin, double XAxisMax, double YAxisMin, double YAxisMax){
+
+    ///Auxiliar variables used in the minMaxLoc function
+    cv::Point MinXLoc;
+    cv::Point MaxXLoc;
+    cv::Point MinYLoc;
+    cv::Point MaxYLoc;
+    cv::Point MinXLocFindZero;
+    cv::Point MaxXLocFindZero;
+    cv::Point MinYLocFindZero;
+    cv::Point MaxYLocFindZero;
+
+    double MinX;
+    double MaxX;
+    double MinY;
+    double MaxY;
+    double MinXFindZero;
+    double MaxXFindZero;
+    double MinYFindZero;
+    double MaxYFindZero;
+
+    int FigW = FigureWidth;
+    int FigH = FigureHeight;
+
+    int NumVecElements = Xdata.rows;
+
+    //Obtain the minimum and maximum values of Xdata
+    minMaxLoc(Xdata,&MinX,&MaxX,&MinXLoc,&MaxXLoc);
+
+    //Obtain the minimum and maximum values of Ydata
+    minMaxLoc(Ydata,&MinY,&MaxY,&MinYLoc,&MaxYLoc);
+
+    if(XAxisMin != 1e8 && XAxisMax != 1e8 && YAxisMin != 1e8 && YAxisMax != 1e8)
+    {
+        MinX = XAxisMin;
+        MaxX = XAxisMax;
+        MinY = YAxisMin;
+        MaxY = YAxisMax;
+    }
+
+    Mat InterpXdata = linearInterpolation(MinX, MaxX, 0, FigW, Xdata);
+    Mat InterpYdata = linearInterpolation(MaxY, MinY, 0, FigH, Ydata);
+
+    //Find the zeros in image coordinates
+    Mat XdataPlusZero = Mat::zeros(Xdata.rows+1,1,CV_64F);
+    Mat YdataPlusZero = Mat::zeros(Ydata.rows+1,1,CV_64F);
+
+    for(int i=0; i<Xdata.rows; i++){
+        XdataPlusZero.at<double>(i,0) = Xdata.at<double>(i,0);
+        YdataPlusZero.at<double>(i,0) = Ydata.at<double>(i,0);
+    }
+
+    minMaxLoc(XdataPlusZero,&MinXFindZero,&MaxXFindZero,&MinXLocFindZero,&MaxXLocFindZero);
+    minMaxLoc(YdataPlusZero,&MinYFindZero,&MaxYFindZero,&MinYLocFindZero,&MaxXLocFindZero);
+
+    if(XAxisMin != 1e8 && XAxisMax != 1e8 && YAxisMin != 1e8 && YAxisMax != 1e8)
+    {
+        MinXFindZero = XAxisMin;
+        MaxXFindZero = XAxisMax;
+        MinYFindZero = YAxisMin;
+        MaxYFindZero = YAxisMax;
+    }
+
+    Mat InterpXdataFindZero = linearInterpolation(MinXFindZero, MaxXFindZero, 0, FigW, XdataPlusZero);
+    Mat InterpYdataFindZero = linearInterpolation(MaxYFindZero, MinYFindZero, 0, FigH, YdataPlusZero);
+
+    double ImageXzero = InterpXdataFindZero.at<double>(NumVecElements,0);
+    double ImageYzero = InterpYdataFindZero.at<double>(NumVecElements,0);
+
+    double CurrentX = Xdata.at<double>(NumVecElements-1,0);
+    double CurrentY = Ydata.at<double>(NumVecElements-1,0);
+
+    Mat FigureRed	= Mat::zeros(FigH, FigW, CV_64F);
+    Mat FigureGreen = Mat::zeros(FigH, FigW, CV_64F);
+    Mat FigureBlue	= Mat::zeros(FigH, FigW, CV_64F);
+    std::vector<cv::Mat> FigureVec;
+    Mat Figure		= Mat::zeros(FigH, FigW, CV_64F);
+
+    //Draw the plot by connecting lines between the points
+    cv::Point p1;
+    p1.x = InterpXdata.at<double>(0,0);
+    p1.y = InterpYdata.at<double>(0,0);
```

Comment:
Change both to

+ p1.x = (int)InterpXdata.at<double>(0,0);
+ p1.y = (int)InterpYdata.at<double>(0,0);

to suppress warnings